Sources/StateGraph/iCloudKVStored.swift

```diff
@@ -0,0 +1,215 @@
+  }
+  @_spi(Internal)
+  public func _setValue(to store: NSUbiquitousKeyValueStore, forKey key: String) {
+    store.set(self, forKey: key)
+    store.synchronize()
+  }
+}
+
+extension Date: iCloudKVStorable {
+  @_spi(Internal)
+  public static func _getValue(from store: NSUbiquitousKeyValueStore, forKey key: String, defaultValue: Date) -> Date {
+    return store.object(forKey: key) as? Date ?? defaultValue
+  }
+  @_spi(Internal)
+  public func _setValue(to store: NSUbiquitousKeyValueStore, forKey key: String) {
+    store.set(self, forKey: key)
+    store.synchronize()
+  }
+}
+
+private enum Static {
+  static let decoder = JSONDecoder()
+  static let encoder = JSONEncoder()
+}
+
+extension iCloudKVStorable where Self: Codable {
+  public static func _getValue(from store: NSUbiquitousKeyValueStore, forKey key: String, defaultValue: Self) -> Self {
+    guard let data = store.data(forKey: key) else {
+      return defaultValue
+    }
+    do {
+      return try Static.decoder.decode(Self.self, from: data)
+    } catch {
+      return defaultValue
+    }
+  }
+  
+  public func _setValue(to store: NSUbiquitousKeyValueStore, forKey key: String) {
+    do {
+      let data = try Static.encoder.encode(self)
+      store.set(data, forKey: key)
+      store.synchronize()
+    } catch {
+      // If encoding fails, remove the key to avoid inconsistent state
+      store.removeObject(forKey: key)
+      store.synchronize()
+    }
+  }
+}
+
+extension Optional: iCloudKVStorable where Wrapped: iCloudKVStorable {
+  @_spi(Internal)
+  public static func _getValue(from store: NSUbiquitousKeyValueStore, forKey key: String, defaultValue: Self) -> Self {
+    if store.object(forKey: key) == nil {
+      return defaultValue
+    }
+    // Try to get the wrapped value using a temporary dummy value
+    let tempValue = store.object(forKey: key)
+    if tempValue == nil {
+      return nil
+    }
+    return tempValue as? Wrapped
+  }
+  @_spi(Internal)
+  public func _setValue(to store: NSUbiquitousKeyValueStore, forKey key: String) {
+    if let value = self {
+      value._setValue(to: store, forKey: key)
+    } else {
+      store.removeObject(forKey: key)
+      store.synchronize()
+    }
+  }
+}
+
+/// A node that functions as an endpoint in a Directed Acyclic Graph (DAG) and persists its value to iCloudKV.
+///
+/// `iCloudKVStored` can have its value set directly from the outside, and changes to its value
+/// automatically propagate to dependent nodes. The value is automatically persisted to iCloudKV
+/// and restored when the node is recreated.
+///
+/// - When value changes: Changes propagate to all dependent nodes and are persisted to iCloudKV
+/// - When value is accessed: Dependencies are recorded and the value is loaded from iCloudKV if needed
+/// - When other devices change the value: The local value is automatically updated
+public typealias iCloudKVStored<Value: iCloudKVStorable> = _Stored<Value, iCloudKVStorage<Value>>
```

Comment:
At least, type name should be PascalCase.
so like how about this - CloudKitKeyValueStored rather than calling them `iCloud`.